### ISSUE TITLE
Sort filter fails in Betterbird

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/quickFilters#151]
Sorting filters doesn't work in the latest version of Betterbird.

Error log:

searchTerm: termIdxFromRichlistIdx 0 -> -1 searchTerm.js:58:15
searchTerm: termIdxFromRichlistIdx 1 -> 0 2 searchTerm.js:70:13
searchTerm: termIdxFromRichlistIdx 2 -> 1 searchTerm.js:70:13
initializeSearchRows: set top-most to or from 1 searchTerm.js:337:17
searchTerm: termIdxFromRichlistIdx 2 -> 1 searchTerm.js:70:13
searchTerm: termIdxFromRichlistIdx 3 -> 2 searchTerm.js:70:13
initializeSearchRows: set top-most to or from 2 searchTerm.js:337:17
searchTerm: termIdxFromRichlistIdx 3 -> 2 searchTerm.js:70:13
searchTerm: termIdxFromRichlistIdx 4 -> 3 searchTerm.js:70:13
initializeSearchRows: set top-most to or from 3 searchTerm.js:337:17
searchTerm: termIdxFromRichlistIdx 4 -> 3 searchTerm.js:70:13
searchTerm: termIdxFromRichlistIdx 5 -> 4 searchTerm.js:70:13
initializeSearchRows: set top-most to or from 4 searchTerm.js:337:17
searchTerm: initializeSearchRows - exit 0: and   level 1 searchTerm.js:44:13
searchTerm: initializeSearchRows - exit 1: or   level 1 searchTerm.js:44:13
searchTerm: initializeSearchRows - exit 2: or   level 1 searchTerm.js:44:13
searchTerm: initializeSearchRows - exit 3: or   level 1 searchTerm.js:44:13
searchTerm: initializeSearchRows - exit 4: or   level 1 searchTerm.js:44:13
TypeError: gFilter.searchTerms[i] is undefined qFilters-filterEditor.js:142:15
ReferenceError: gTotalSearchTerms is not defined qFilters-filterEditor.js:391:14

**Comment by RealRaven2000** [comment]
This may be due to a missing (or differently named) variable [gTotalSearchTerms]. In the Thunderbird codebase this is defined here:

https://searchfox.org/comm-central/rev/5fb08affe686ce86e49678f02d82fcce06370525/mailnews/search/content/searchTerm.js#7

**Comment by Lab-doc** [comment]
It seems there is a bigger issue: Clicking Sort deletes the items in the filter search list. This occurs in the previous Betterbird release and current release (just posted): 102.7.0-bb28 (64-bit)

**Comment by TonyGravagno** [comment]
This seems to be related, from the current Tb (not Betterbird) and quickFilters :  

![image](https://github.com/user-attachments/assets/6cf7b775-a298-4da1-81c2-accc090a4685)

gFilters-filterEditor.js:140 :  

`if (gFilter.searchTerms[i].attrib == -2) {`

What's weird is that it's complaining about gFilter not being defined rather than the searchTerms within.

Should I move this to a new ticket?

**Comment by RealRaven2000** [comment]
> What's weird is that it's complaining about gFilter not being defined rather than the searchTerms within.
> 
> Should I move this to a new ticket?

yes, please log the bug and whether the sorting is broken in Thunderbird. I am simply not sure at this stage whether the bug still exists in Betterbird. need to test when I am back home.
